lcm/src/lcm/providers/udpm.rs

```diff
@@ -181,6 +181,7 @@ impl<'a> UdpmProvider<'a> {
     pub fn unsubscribe(&mut self, subscription: Subscription) {
         self.subscriptions
             .retain(|&(ref sub, _)| *sub != subscription);
+        drop(subscription);
```

Comment:
Why was this added? Subscription is already dropped at the end of the function (and it's really just an integer, so dropping doesn't really do anything).

Reply:
Since the function didn't consume `subscription` clippy recommended that the function take a reference instead. Calling drop to explicitly consume the variable silences the warning and ensures that the caller can't hold on to an invalid subscription.